From: Piotr Kubik <piotr.kubik@adtran.com>
Date: Wed, 26 Mar 2025 19:03:53 +0100
Subject: [PATCH] net: pse-pd: Fix mdiobus_register when no PSE

Prior to this patch when no PSE controllers were registered at
mdiobus_register() call, MDIO bus registration failed and no MDIO ports
were operational.

This can happen when PSE controller driver is compiled as a module and
loaded after a module registering MDIO bus.

This is not ideal solution because loading PSE controller module
after registering MDIO bus will not make the PoE operational
due to missing PoE initialization done in mdiobus_register(),
but this patch at least makes MDIO ports functional in this case.

Signed-off-by: Piotr Kubik <piotr.kubik@adtran.com>
---
 drivers/net/pse-pd/pse_core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/drivers/net/pse-pd/pse_core.c
+++ b/drivers/net/pse-pd/pse_core.c
@@ -1942,7 +1942,7 @@ struct pse_control *of_pse_control_get(s
 	}
 
 	if (!pcdev) {
-		psec = ERR_PTR(-EPROBE_DEFER);
+		psec = ERR_PTR(-ENOENT);
 		goto out;
 	}
 
